OptimoveSDK/Sources/Classes/PreferenceCenter/OptimovePreferenceCenter.swift

```diff
@@ -0,0 +1,198 @@
+import Foundation
+import OptimoveCore
+
+@available(iOS 13.0, *)
+public class OptimovePreferenceCenter {
+    enum Error: LocalizedError {
+        case alreadyInitialized
+        case configurationIsMissing
+
+        var errorDescription: String? {
+            switch self {
+            case .alreadyInitialized:
+                return "The PreferenceCenterSDK has already been initialized."
+            case .configurationIsMissing:
+                return "Preference center config is missing."
+            }
+        }
+    }
+
+    private static var instance: OptimovePreferenceCenter?
+    private var config: PreferenceCenterConfig?
```

Comment:
1 comment i had in android recently, is that if delayed initialisation is updaing OptimoveConfig, we need to make sure PreferencCenterConfig gets updated too. to not breed copies/references, can store OptimoveConfig here